stories/organisms/SelectedItemBadge.stories.tsx

```diff
@@ -132,7 +132,57 @@ export const _SelectedItemBadgeMultiSelect: Story = {
         );
         const [selectedSynonyms, setSelectedSynonyms] = React.useState<string[]>([]);
 
-        const handleOnChange = (synonym: string) => {
+        React.useEffect(() => {
+            if (
+                selectedSynonyms.includes('All') &&
+                selectedSynonyms.length !== synonyms.length + synonyms2.length + 2
+            ) {
+                setSelectedSynonyms(selectedSynonyms.filter((s) => s !== 'All'));
+            }
+
+            if (
+                !selectedSynonyms.includes('All') &&
+                selectedSynonyms.length === synonyms.length + synonyms2.length + 1
+            ) {
+                setSelectedSynonyms([...selectedSynonyms, 'All']);
+            }
+        }, [selectedSynonyms]);
+
+        const handleOnChange = (event: Event, synonym: string) => {
+            // @ts-ignore
+            const checked = !Array.from(event.target.classList).includes(
```

Comment:
do we really need to rely on DOM manipulations? Maybe we could just use selectedSynonyms state to check if it is checked/selected or not? What do you think?

Reply:
I was not aware of the onCheckedChange